docs/standard/base-types/backtracking-in-regular-expressions.md

```diff
@@ -198,6 +205,16 @@ If you do not set a time-out value explicitly, the default time-out value is det
 |`[A-Z]\w*`|Match an alphabetical character followed by zero or more word characters.|
 |`$`|End the match at the end of the input string.|
 
+### General Performance Considerations
+
+The following suggestions are not specifically to prevent excessive backtracking, but may help increase the performance of your regular expression:
+
+1. If a pattern will be heavily used, precompile it. The best way to do this is to use the [regular expression source generator](regular-expression-source-generators.md) to precompile it. If the source generator is not available for your app, use the <xref:System.Text.RegularExpressions.RegexOptions.Compiled?displayProperty=nameWithType> option.
+
+1. If matches are known to always start beyond a certain offset into the pattern, consider passing the offset to the regular expression engine. For example, using the overload <xref:System.Text.RegularExpressions.Regex.Match%28System.String%2CSystem.Int32%29?displayProperty=nameWithType>. This will reduce the amount of the text the engine needs to consider.
+
+1. If matches are typically found near the end of a very large input, <xref:System.Text.RegularExpressions.RegexOptions.RightToLeft?displayProperty=nameWithType> might find it more quickly. However, use this with care. The regular expression engines are most heavily optimized for left to right operation, so you should test to ensure this is an improvement. It may be better to pass in the offset instead.
+
```

Comment:
Other tips you could elaborate on:
- For situations where you'll be performing matches with the same pattern repeatedly, rather than using the static Regex methods or creating and throwing away a Regex instance, create a Regex instance and store it for reuse.  Constructing Regex objects is relatively expensive.  The source generator implicitly caches.
- Use the method best suited for the information you need.  For example, if you only need to know whether a match occurs but not where the match occurs, prefer using IsMatch.  If you only need to know how many times something matches, prefer using Count. If you only need to know the bounds of a match but not anything about a match's captures, prefer using EnumerateMatches rather than Match or Matches.  The more information the engine needs to provide, the more expensive the operation will be.
- Reduce unnecessary captures.  If you want to use parentheses in your patterns but don't actually need the resulting captures, prefer to use either non-capturing groups or specify RegexOptions.ExplicitCapture.